### PR TITLE
Test/dynamite/petstore

### PR DESCRIPTION
**Pull request by Leptopoda**
resolves: #761 
I've tried setting up the petstore docker container form the petstore repo (https://github.com/swagger-api/swagger-petstore) but I run into java killing itself with:
```
library initialization failed - unable to allocate file descriptor table - out of memory%                                                                                                          
```

According to this [stackoverflow](https://stackoverflow.com/questions/49435109/error-upon-jar-execution-unable-to-allocate-file-descriptor-table#56895801) thread this is because of the number of open files and it should be fixed with java 9.
I don't have time to further investigate this (build the docker container with a newer java version) and would rather keep my high open file count (as I regularly need this). If anyone wants to pick this up feel free :)

### CODE REVIEW

File: packages/dynamite/dynamite/lib/src/builder/resolve_mime_type.dart [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                ..name = parameterName
+                ..type = refer(result.nullableName)
+                ..named = true
+                ..required = dartParameterRequired,
+            ),
+          );
+
+          if (dartParameterNullable) {
+            code.write('if ($parameterName != null) {');
+          }
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/resolve_mime_type.dart:135]: Can we unify this with the code above?

File: packages/dynamite/dynamite/lib/src/builder/resolve_mime_type.dart [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        state,
+        'identifier-$mimeType',
+        mediaType.schema!,
+      );
+
+      if (mimeType == '*/*' || mimeType == 'application/octet-stream' || mimeType.startsWith('image/')) {
+        return ('Uint8List', '_response.bodyBytes', true);
+      } else if (mimeType.startsWith('text/') || mimeType == 'application/javascript') {
+        return ('String', '_response.body', true);
+      } else if (mimeType == 'application/json') {
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/builder/resolve_mime_type.dart:37]: This could be turned into a switch statement

Comment (Leptopoda) [reply packages/dynamite/dynamite/lib/src/builder/resolve_mime_type.dart:37]: It has `mimeType.startsWith()` and I don't think that we could put it in a switch

**Comment by provokateurin** [comment]
I think we should generate the e2e test as the example inside the dynamite package.

**Comment by Leptopoda** [comment]
Closing in favor of a new version @provokateurin  is working on

**Pull request closed without merging**